File: exercise/week04/php/src/Construct/Artefact.php
<?php

namespace Construct;

class Artefact
{
    public string $name;
    public int $timeToLive;
    public int $integrity;

    public function __construct(string $name, int $timeToLive, int $integrity)
    {
        $this->name = $name;
        $this->timeToLive = $timeToLive;
        $this->integrity = $integrity;
    }

    public function __toString(): string
    {
        return "{$this->name}, 🥄 integrity: {$this->integrity}, 💓 timeToLive: {$this->timeToLive}";
    }
}

File: exercise/week04/c#/Construct/Artefact.cs
namespace Construct;

public class Artefact(string name, int timeToLive, int integrity)
{
    public string Name { get; set; } = name;
    public int TimeToLive { get; set; } = timeToLive;
    public int Integrity { get; set; } = integrity;

    public override string ToString() => $"{Name}, 🥄 integrity: {Integrity}, 💓 timeToLive: {TimeToLive}";
}